tests/apps/courses/test_api_course_run_sync_edx.py

```diff
@@ -970,3 +987,129 @@ def test_api_course_run_sync_enrollment_count(self, mock_signal):
+        self.assertEqual(json.loads(response.content), {"success": True})
+        course_run.refresh_from_db()
+        self.assertEqual(
+            course_run.catalog_visibility, CourseRunCatalogVisibility.COURSE_ONLY
+        )
+        self.assertFalse(mock_signal.called)
+
+    @override_settings(TIME_ZONE="utc")
+    def test_api_course_run_sync_catalog_visibility_course_and_search(
+        self, mock_signal
+    ):
+        """
+        Verify that the catalog visibility is updated with `course_and_search`
+        """
+        link = "http://example.edx:8073/courses/course-v1:edX+DemoX+01/course/"
+        course = CourseFactory(code="DemoX")
+        course_run = CourseRunFactory(direct_course=course, resource_link=link)
+        mock_signal.reset_mock()
+
+        data = {
+            "resource_link": link,
+            "catalog_visibility": "course_and_search",
+        }
+
+        response = self.client.post(
+            "/api/v1.0/course-runs-sync",
+            data,
+            content_type="application/json",
+            HTTP_AUTHORIZATION=(
+                "SIG-HMAC-SHA256 84583c2b96759c26ed1ecb3fc7d6d4805853cbbcd5ac40e777c0aa1c319fd490"
+            ),
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(json.loads(response.content), {"success": True})
+        course_run.refresh_from_db()
+        self.assertEqual(
+            course_run.catalog_visibility, CourseRunCatalogVisibility.COURSE_AND_SEARCH
+        )
+        self.assertFalse(mock_signal.called)
+
+    @override_settings(TIME_ZONE="utc")
+    def test_api_course_run_sync_catalog_visibility_hidden(self, mock_signal):
+        """
+        Verify that the catalog visibility is updated with `hidden`
+        """
+        link = "http://example.edx:8073/courses/course-v1:edX+DemoX+01/course/"
+        course = CourseFactory(code="DemoX")
+        course_run = CourseRunFactory(direct_course=course, resource_link=link)
+        mock_signal.reset_mock()
+
+        data = {
+            "resource_link": link,
+            "catalog_visibility": "hidden",
+        }
+
+        response = self.client.post(
+            "/api/v1.0/course-runs-sync",
+            data,
+            content_type="application/json",
+            HTTP_AUTHORIZATION=(
+                "SIG-HMAC-SHA256 b18aaa037ac3c4630fb3d88597cf5138e8c145a10b912688de82ee2c963a7ed6"
+            ),
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(json.loads(response.content), {"success": True})
+        course_run.refresh_from_db()
+        self.assertEqual(
+            course_run.catalog_visibility, CourseRunCatalogVisibility.HIDDEN
+        )
+        self.assertFalse(mock_signal.called)
+
+    @override_settings(TIME_ZONE="utc")
+    def test_api_course_run_sync_catalog_visibility_none(self, mock_signal):
+        """
+        Verify that the catalog visibility is updated with `none` value
```

Comment:
ditto

Reply:
actually this is redundant with the above test?

Reply:
This is already ok. I've included one test to test the adapt from open edx to Richie values. Probably I should include others cases like `both` and `about`.

Reply:
I included 6 tests: The first 3 test if the catalog visibility has the Richie values, the last 3 tests the adaptation of Open edX to Richie values.
- test_api_course_run_sync_catalog_visibility_course_only
- test_api_course_run_sync_catalog_visibility_course_and_search
- test_api_course_run_sync_catalog_visibility_hidden
- test_api_course_run_sync_catalog_visibility_none
- test_api_course_run_sync_catalog_visibility_both
- test_api_course_run_sync_catalog_visibility_about